### PR TITLE
MIP-414 PR

### PR DESCRIPTION
**Pull request by kkingavio**
Sorry for number of files changed... 

MunitMavenPlugin Attributes Rules... 

Added rule, tests... Moved MuleXmlParser to different package, renamed PomProperty to PomElement

### CODE REVIEW

File: src/main/groovy/com/avioconsulting/mule/linter/model/pom/MunitMavenPlugin.groovy [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    Integer getIgnoreFilesLineNo() {
+        MuleXmlParser.getNodeLineNumber(pluginXml.configuration.coverage.ignoreFiles)
+    }
+
+}
+
+/*
+			<plugin>
```
Comment (adesjardin) [inline src/main/groovy/com/avioconsulting/mule/linter/model/pom/MunitMavenPlugin.groovy:44]: If we want to keep a sample in here put it in a groovydoc on the class

File: src/main/groovy/com/avioconsulting/mule/linter/model/pom/PomFile.groovy [hunk 2]
```diff
@@ -59,8 +61,31 @@ class PomFile extends ProjectFile {
...
+        }
+        return plugin
+    }
+
+    MunitMavenPlugin getMunitPlugin() {
+        PomPlugin pp = getPlugin(MunitMavenPlugin.GROUP_ID, MunitMavenPlugin.ARTIFACT_ID)
+        return pp == null ? null : new MunitMavenPlugin(pp.pluginXml, this)
+    }
+
     private GPathResult getPomProperties() {
```
Comment (adesjardin) [inline src/main/groovy/com/avioconsulting/mule/linter/model/pom/PomFile.groovy:81]: This isn't used and we discussed not exposing the full properties GPathResult

File: src/main/groovy/com/avioconsulting/mule/linter/model/pom/PomFile.groovy [hunk 3]
```diff
@@ -59,8 +61,31 @@ class PomFile extends ProjectFile {
...
         return pomXml[PROPERTIES] as GPathResult
     }
 
+    @SuppressWarnings('DuplicateStringLiteral')
+    private String getPropertyValueFromVariable(String variable) {
```
Comment (adesjardin) [inline src/main/groovy/com/avioconsulting/mule/linter/model/pom/PomFile.groovy:86]: This isn't used anymore

File: src/main/groovy/com/avioconsulting/mule/linter/rule/pom/MunitMavenPluginAttributesRule.groovy [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    @Override
+    List<RuleViolation> execute(Application application) {
+        List<RuleViolation> violations = []
+        MunitMavenPlugin plugin = application.pomFile.munitPlugin
+        if (plugin != null) {
+            coverageAttributeMap.each { key, val ->
+                PomElement pe = plugin.getConfigProperty(key)
+                if (pe == null || pe.value != val) {
```
Comment (adesjardin) [inline src/main/groovy/com/avioconsulting/mule/linter/rule/pom/MunitMavenPluginAttributesRule.groovy:53]: use pe?.value to simplify

File: src/main/groovy/com/avioconsulting/mule/linter/rule/pom/MunitMavenPluginAttributesRule.groovy [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        } else {
+            violations.add(new RuleViolation(this, PomFile.POM_XML, 0, 'Missing munit-maven-plugin'))
+        }
+        return violations
+    }
+
+}
+
+//@Override
+//List<RuleViolation> execute(Application application) {
```
Comment (adesjardin) [inline src/main/groovy/com/avioconsulting/mule/linter/rule/pom/MunitMavenPluginAttributesRule.groovy:74]: Remove commented code

File: src/main/groovy/com/avioconsulting/mule/linter/model/pom/PomFile.groovy (hunk 2 above)
Comment (kkingavio) [reply src/main/groovy/com/avioconsulting/mule/linter/model/pom/PomFile.groovy:81]: the getPomProperties?  that method is private, and only used by getPomProperty... 

I hadn't changed the getPomProperties...

Comment (kkingavio) [reply src/main/groovy/com/avioconsulting/mule/linter/model/pom/PomFile.groovy:81]: Used by 2 methods (getPomProperty and getPropertyLineNo)

File: src/main/groovy/com/avioconsulting/mule/linter/model/pom/MunitMavenPlugin.groovy (hunk 1 above)
Comment (kkingavio) [reply src/main/groovy/com/avioconsulting/mule/linter/model/pom/MunitMavenPlugin.groovy]: Done

File: src/main/groovy/com/avioconsulting/mule/linter/model/pom/PomFile.groovy (hunk 3 above)
Comment (kkingavio) [reply src/main/groovy/com/avioconsulting/mule/linter/model/pom/PomFile.groovy]: Yup, removed

File: src/main/groovy/com/avioconsulting/mule/linter/rule/pom/MunitMavenPluginAttributesRule.groovy (hunk 5 above)
Comment (kkingavio) [reply src/main/groovy/com/avioconsulting/mule/linter/rule/pom/MunitMavenPluginAttributesRule.groovy]: :+1:

File: src/main/groovy/com/avioconsulting/mule/linter/rule/pom/MunitMavenPluginAttributesRule.groovy (hunk 4 above)
Comment (kkingavio) [reply src/main/groovy/com/avioconsulting/mule/linter/rule/pom/MunitMavenPluginAttributesRule.groovy]: :+1:

**Pull request merged**